content/v2.0/actions/mime-types-and-format.md

```diff
@@ -0,0 +1,170 @@
+---
+title: MIME Types and format
+order: 70
+---
+```ruby
+module MyApp
+  class App < Hanami::App
+    config.actions.format json: "application/json"
+    config.actions.default_request_format = :json
+    config.actions.default_response_format = :json
+    config.middleware.use :body_parser, :json
+  end
+end
+```
+
+
+Actions have advanced features for MIME Type detection, automatic headers, whitelisting etc..
+
+## Request Introspection
+
+In order to understand what the requested MIME Type is, an action looks at the `Accept` request header and exposes a high level API: `.format`.
+
+The `format` method accepts as an argument a symbol representation of the MIME Type (eg. `:html`, `:json`, `:xml` etc..).
+
+```ruby
+# app/actions/books/index.rb
+
+module Bookshelf
+  module Actions
+    module Books
+      class Index < Bookshelf::Action
+        format :json
+        
+        def handle(*, response)
+          response.body = [{foo: 'bar'}].to_json
+        end
+      end
+    end
+  end
+end
+```
+
+This will always set the `Content-Type` to `application/json` on the response headers.
+
+## Automatic Content-Type
+
+If the explicit format is not defined
+An action returns the `Content-Type` response header automatically according to the requested MIME Type and charset.
+
+If the client asks for `Accept: text/html,application/xhtml+xml,application/xml;q=0.9`, the action will return `Content-Type: text/html; charset=utf-8`.
+
+### Default Charset
+
+In case you would like o change the default charset for your action, We can specify a different default charset to return.
```

Comment:
```suggestion
In case you would like to change the default charset for your action, you can specify a different default charset to return using the `default_charset` method.
```